test/tests/functional/pbs_basil_support.py

```diff
@@ -362,5 +407,393 @@ def test_hbmemm_rsc(self):
+
+    def test_pbs_alps_in_sync(self):
+        """
+        Check for the presence of message indicating PBS and ALPS are
+        in sync.
+        """
+        # Determine if BASIL 1.7 is supported.
+        rv = self.mom.log_match(
+                "This Cray system supports the BASIL 1.7 protocol.", n='ALL')
+        if not rv:
+            self.skip_test('Skipping this test because there are no KNL nodes present.')
+            return
+
+        # Determine if KNL vnodes are present.
+        knl_vnodes = self.get_knl_vnodes()
+
+        if len(knl_vnodes) == 0:
+            self.skipTest(reason='No KNL vnodes present')
+            return
+        else:
+            self.logger.info("KNL vnode list: %s" % (knl_vnodes))
+
+        # Check for PBS ALPS Inventory Hook message.
+        now = time.time()
+        rv = self.mom.log_match("ALPS Inventory Check: PBS and ALPS are in sync",
+                                starttime=(now), max_attempts=31, interval=10)
+        if rv:
+            self.logger.info('Expected ALPS Inventory Hook message found.')
+        else:
+            self.logger.info('Expected ALPS Inventory Hook message not found.')
+        self.assertTrue(rv)
+
+    def test_knl_batch_to_interactive(self):
+        """
+        Change the mode of any two KNL nodes to interactive. Then check if the
+        PBS_alps_inventory_check hook picks up on the change and nodes are
+        marked as stale. Restore changes to hook and mode of KNL nodes.
+        """
+        if self.du.platform == 'craysim':
+            self.skipTest(reason='xtprocadmin cmd is not on cray simulator')
+
+        # Find the list of KNL vnodes
+        klist = self.retklist()
+        if len(klist) == 0:
+            self.skipTest(reason='No KNL vnodes present')
+            return
+        else:
+            self.logger.info("KNL vnode list: %s" % (klist))
+
+        # Set pbshook frequency to 10 seconds
+        self.server.manager(MGR_CMD_SET, PBS_HOOK,
+                            {'enabled': 'true', 'freq': 10},
+                            id='PBS_alps_inventory_check', expect=True)
+
+        now = time.time()
+        # Change mode of two KNL nodes to interactive
+        if len(klist) >= 2:
+            k1 = klist[0]
+            k2 = klist[len(klist) - 1]
+            knl1 = int(re.search(r'\d+', k1).group())
+            knl2 = int(re.search(r'\d+', k2).group())
+
+        cmd = ['xtprocadmin', '-k', 'm', 'interactive', '-n', "%s" % (knl1)]
+        self.logger.info(cmd)
+        ret = self.server.du.run_cmd(self.server.hostname,
+                                     cmd, logerr=True)
+        self.assertEqual(ret['rc'], 0)
+
+        cmd = ['xtprocadmin', '-k', 'm', 'interactive', '-n', "%s" % (knl2)]
+        self.logger.info(cmd)
+        ret = self.server.du.run_cmd(self.server.hostname,
+                                     cmd, logerr=True)
+        self.assertEqual(ret['rc'], 0)
+
+        # Restart Mom with signal HUP
```

Comment:
You aren't restarting mom with a HUP.  You're forcing her to reread her config file and reread the inventory.  I suspect you're interested in the rereading of the inventory.